src/main/java/org/killbill/billing/client/KillBillHttpClient.java

```diff
@@ -117,12 +118,21 @@ public KillBillHttpClient(final String kbServerUrl, final String username, final
         this.apiKey = apiKey;
         this.apiSecret = apiSecret;
 
-
         final AsyncHttpClientConfig.Builder cfg = new AsyncHttpClientConfig.Builder();
 
+        if(requestTimeout != null)
+        {
+            cfg.setRequestTimeout(requestTimeout);
+            requestTimeoutSec = (int) Math.ceil((double) requestTimeout / 1000);
```

Comment:
Maybe we should always add `1` if `TimeUnit.SECONDS.toMillis(TimeUnit.MILLISECONDS.toSeconds(requestTimeout)) != requestTimeout `. It would solve the 0 problem you mention and also the condition where it becomes lower than the HTTP request timeout.